aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -99,29 +102,40 @@ static class DefaultAwsClientFactory implements AwsClientFactory {
     public S3Client s3() {
       return S3Client.builder()
           .httpClientBuilder(configureHttpClientBuilder(httpClientType))
-          .applyMutation(builder -> configureEndpoint(builder, s3Endpoint))
+          .applyMutation(builder -> configureRegion(builder, s3Endpoint))
+          .applyMutation(builder -> configureEndpoint(builder, region))
           .serviceConfiguration(s3Configuration(s3PathStyleAccess, s3UseArnRegionEnabled))
           .credentialsProvider(credentialsProvider(s3AccessKeyId, s3SecretAccessKey, s3SessionToken))
           .build();
     }
 
     @Override
     public GlueClient glue() {
-      return GlueClient.builder().httpClientBuilder(configureHttpClientBuilder(httpClientType)).build();
+      return GlueClient.builder()
+          .httpClientBuilder(configureHttpClientBuilder(httpClientType))
+          .applyMutation(builder -> configureEndpoint(builder, region))
```

Comment:
I don't think this is right, we're calling configureEndpoint with the region, instead of the new configureRegion?

Reply:
Yeah. It's my bad. fixed.